packages/components/src/combobox-control/index.tsx

```diff
@@ -187,14 +195,15 @@ function ComboboxControl( {
 	};
 
 	const handleOnReset = () => {
-		setValue( null );
-		inputContainer.current.focus();
+		setValue( undefined );
+		inputContainer.current?.focus();
 	};
 
 	// Update current selections when the filter input changes.
 	useEffect( () => {
 		const hasMatchingSuggestions = matchingSuggestions.length > 0;
 		const hasSelectedMatchingSuggestions =
+			selectedSuggestion &&
 			matchingSuggestions.indexOf( selectedSuggestion ) > 0;
```

Comment:
If `selectedSuggestion` is falsy, this can't look for it in `matchingSuggestions`